dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -0,0 +1,1374 @@
+package org.bitcoins.dlc.wallet
+
+import org.bitcoins.core.api.chain.ChainQueryApi
+import org.bitcoins.core.api.feeprovider.FeeRateApi
+import org.bitcoins.core.api.node.NodeApi
+import org.bitcoins.core.api.wallet.db._
+import org.bitcoins.core.config.BitcoinNetwork
+import org.bitcoins.core.crypto.ExtPublicKey
+import org.bitcoins.core.currency._
+import org.bitcoins.core.hd._
+import org.bitcoins.core.number._
+import org.bitcoins.core.protocol._
+import org.bitcoins.core.protocol.dlc.build.DLCTxBuilder
+import org.bitcoins.core.protocol.dlc.models.DLCMessage.DLCAccept._
+import org.bitcoins.core.protocol.dlc.models.DLCMessage._
+import org.bitcoins.core.protocol.dlc.models.DLCStatus._
+import org.bitcoins.core.protocol.dlc.models._
+import org.bitcoins.core.protocol.dlc.sign._
+import org.bitcoins.core.protocol.script._
+import org.bitcoins.core.protocol.tlv._
+import org.bitcoins.core.protocol.transaction._
+import org.bitcoins.core.util.FutureUtil
+import org.bitcoins.core.wallet.fee.SatoshisPerVirtualByte
+import org.bitcoins.core.wallet.utxo._
+import org.bitcoins.crypto._
+import org.bitcoins.dlc.wallet.internal._
+import org.bitcoins.dlc.wallet.models._
+import org.bitcoins.keymanager.bip39.BIP39KeyManager
+import org.bitcoins.wallet.config.WalletAppConfig
+import org.bitcoins.wallet.{Wallet, WalletLogger}
+import scodec.bits.ByteVector
+
+import java.time.Instant
+import scala.concurrent.{ExecutionContext, Future}
+
+/** A [[Wallet]] with full DLC Functionality */
+abstract class DLCWallet
+    extends Wallet
+    with AnyDLCHDWalletApi
```

Comment:
Could you file an issue detailing the proposed follow-up wallet refactor?

Reply:
#3139